lib/block_view_stake.go

```diff
@@ -547,15 +547,23 @@ func (txindexMetadata *UnlockStakeTxindexMetadata) GetEncoderType() EncoderType
 // DB UTILS
 //
 
-func DBKeyForStakeByValidatorAndStaker(stakeEntry *StakeEntry) []byte {
```

Comment:
Passing in stakeEntry here gives the impression that the caller is passing in a valid stakeEntry, when in reality the caller sometimes passes in a partially constructed entry with nil StakeAmountNanos and default RestakeRewards values.

This is risky and can lead to bugs down the line if we ever need to update this index to use the above, inadvertently thinking the entire stakeEntry is full constructed. 

It's cleaner to pass in the exact parameters needed instead.